Add word click popup feature on Study tab

PFKimmerle opened the pull request:
Fixes #257

## Summary
- User can click on any word to show the popup
- Popup positioned relative to the clicked word so the clicked word is still visible
- Information from the dictionary to display the characters, meaning and pronunciation on the popup
- Popup contain an "X" to close the popup when clicked
- User can also close the popup by clicking anywhere outside the popup
- Popup contains an "+" circle button (not clickable yet) 

Side note,  in some cases the dictionary may return alternate definitions (such as place names). This reflects the current data available and is expected behavior for now.

![screenshot1](https://github.com/user-attachments/assets/fb872555-98fd-4bc6-8642-d72332b84c1d)
![screenshot2](https://github.com/user-attachments/assets/d25d0f48-eadd-4cab-a63d-3d151d1f6fb2)
![screenshot3](https://github.com/user-attachments/assets/772afea8-8797-4f2c-b93d-f217d1248881)
![screenshot4](https://github.com/user-attachments/assets/70903c30-cb80-4226-8f5b-6c572e230252)

PFKimmerle committed:
Add popup on word click on the Study tab

- Added WordPopup to show characters and meaning when clicking words in Study tab
- Positioned popup based on click location
- "+" button included

PFKimmerle committed:
updated

- updated to center popup on word

JPEscobari reviewed (changes requested):
Thank you so much for taking care of all of this @PFKimmerle 

I'm in your branch and can verify our pop-up is working correctly.

Now, sorry for not mentioning this in the issue's description, but do you think you can add a subtle hover word color effect, as show in the `Demo Study Tab`?

I believe the color is `#ddfcf6`, but it's just a tad darker than what you have in `StudyTab.scss` for the class `.study_word--active{ }`

The screenshot here shows the expected hover effect that should come up when the user's cursor is over the word:
![image](https://github.com/user-attachments/assets/9291bd9e-e01e-4c9e-ba0a-4ed10fe4fa4d)

This should align our word selection UX with what's shown in the Demo.

Let me know if you have questions about this final detail or if you'll need an extra hand making these changes.

PFKimmerle committed:
Updated StudyTab and WordPopup components

- Added subtle hover effect for word highlights
- Removed the previous active word background color
- Updated popup border color for consistency

PFKimmerle commented:
@JPEscobari Thanks for the review and the image reference. I made the suggested changes and updated the StudyTab and WordPopup components:

- Added subtle hover effect for word highlights
- Removed the previous active word background color
- Updated popup border color for consistency

Let me know if you'd like me to adjust anything else.

![screenshot6](https://github.com/user-attachments/assets/1742c871-d601-409a-a1dd-e6724a6cf34a)

JPEscobari reviewed (approved):
There you go! Massive contribution to the project @PFKimmerle , as this feature opens the door for issue #258 and beyond.

Thank you and let's keep pushing forward!
